libvast/test/system/pivoter.cpp

```diff
@@ -0,0 +1,143 @@
+1258531924.321413	CojK5e1MpFgJnwlp6a	192.168.1.103	68	192.168.1.1	67	udp	dhcp	0.044779	303	300	SF	-	-	0	Dd	1	331	1	328	-	1:oG55uQUH+XuHYHOFV0c+yOutW8E=
+1258531939.613071	Cgq2X52qyXxAAi9avc	192.168.1.102	138	192.168.1.255	138	udp	-	-	-	-	S0	-	-	0	D	1	229	0	0	-	1:4iHhzk49NeoFdK6VHSCw4ruRbsw=
+1258532046.693816	CwF9px2owZEPahqWsg	192.168.1.104	68	192.168.1.1	67	udp	dhcp	0.002103	311	300	SF	-	-	0	Dd	1	339	1	328	-	-
+1258532143.457078	Cu0QHL1w6Dp3Z5y0Pg	192.168.1.102	1170	192.168.1.1	53	udp	dns	0.068511	36	215	SF	-	-	0	Dd	1	64	1	243	-	1:FVMx3YawO69eZmiaMJJbrs6447E=
+1258532203.657268	C37rla3neljCJgPsE2	192.168.1.104	1174	192.168.1.1	53	udp	dns	0.170962	36	215	SF	-	-	0	Dd	1	64	1	243	-	1:79fDvfGNCWV1JBYjXCE5Ov1FuMM=
+1258532331.365294	C4ik1w2JPTX1zO2Ubi	192.168.1.1	5353	224.0.0.251	5353	udp	dns	0.100381	273	0	S0	-	-	0	D	2	329	0	0	-	1:aGi0Bt5ApW6HEEO7wfz+PwvniIU=
+1258532331.365330	CCGBMdPGplqLzQCjg	fe80::219:e3ff:fee7:5d23	5353	ff02::fb	5353	udp	dns	0.100371	273	0	S0	-	-	0	D	2	369	0	0	-	1:JoBDvaK4Tt6BfWSKWPKaJTELr2M=)__";
+
+template <class Reader>
+std::vector<table_slice_ptr> inhale(const char* data) {
+  auto input = std::make_unique<std::istringstream>(data);
+  Reader reader{vast::defaults::system::table_slice_type, caf::settings{},
+                std::move(input)};
+  std::vector<table_slice_ptr> slices;
+  auto add_slice
+    = [&](table_slice_ptr ptr) { slices.emplace_back(std::move(ptr)); };
+  auto [err, produced] = reader.read(std::numeric_limits<size_t>::max(),
+                                     vast::defaults::system::table_slice_size,
+                                     add_slice);
+  if (err != caf::none && err != vast::ec::end_of_input) {
+    auto str = to_string(err);
+    FAIL("reader returned an error: " << str);
+  }
+  return slices;
+}
+
+struct mock_node_state {
+  std::vector<invocation> invocs;
+  static inline constexpr const char* name = "mock-node";
+};
+
+using mock_node_actor = caf::stateful_actor<mock_node_state>;
+
+caf::behavior mock_node(mock_node_actor* self) {
+  return {[=](invocation invocation) {
+    self->state.invocs.push_back(std::move(invocation));
+  }};
+}
+
+struct fixture : fixtures::deterministic_actor_system {
+  fixture() {
+    MESSAGE("spawn mock node");
+    node = sys.spawn(mock_node);
+    run();
+  }
+
+  ~fixture() {
+    self->send_exit(aut, caf::exit_reason::user_shutdown);
+  }
+
+  void spawn_aut(expression expr, std::string target_type) {
+    aut = sys.spawn(vast::system::pivoter, node, std::move(target_type),
+                    std::move(expr));
+    run();
+  }
+
+  const std::vector<table_slice_ptr> slices
+    = inhale<vast::format::zeek::reader>(zeek_conn_m57_head);
+
+  caf::actor node;
+  caf::actor aut;
+};
+
+} // namespace
+
+FIXTURE_SCOPE(pivoter_tests, fixture)
+
+TEST(count IP point query without candidate check) {
+  MESSAGE("build expression");
+  auto expr = unbox(vast::to<expression>("proto == udp && orig_bytes < 600"));
```

Comment:
Do we really need the qualification `vast::` here?

Reply:
I added `using namespace vast;` to the file.